common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java

```diff
@@ -39,36 +43,63 @@
 @RequiredArgsConstructor
 public class AuthService
 {
-    static final String ENVIRONMENT_KEY_HEADER = "hxp-environment";
-    public static final String HXI_AUTH_PROVIDER = "hyland-experience";
+    public static final String HXP_ENVIRONMENT_HEADER = "hxp-environment";
+    public static final String HXP_APP_HEADER = "hxp-app";
+    private static final String HXP_APP_VALUE = "hxai-discovery";
+    public static final String HXP_AUTH_PROVIDER = "hyland-experience";
     public static final String ALFRESCO_AUTH_PROVIDER = "alfresco";
     static final String BASIC = "Basic ";
     static final String BEARER = "Bearer ";
+
     private final AuthProperties authProperties;
     private final AccessTokenProvider accessTokenProvider;
 
     public void setAlfrescoAuthorizationHeaders(Exchange exchange)
     {
-        AuthProperties.AuthProvider authProvider = authProperties.getProviders().get(ALFRESCO_AUTH_PROVIDER);
-        String authType = authProvider.getType();
+        String authType = authProperties.getProviders().get(ALFRESCO_AUTH_PROVIDER).getType();
         clearAuthHeaders(exchange);
-        exchange.getIn().setHeader(AUTHORIZATION, getAlfrescoAuthHeader(authProvider));
+        getAlfrescoAuthHeaders().forEach((key, value) -> exchange.getIn().setHeader(key, value));
         log.debug("Authorization :: {} {} authorization header added", ALFRESCO_AUTH_PROVIDER, authType);
     }
 
     public void setHxIAuthorizationHeaders(Exchange exchange)
     {
-        AuthProperties.AuthProvider authProvider = authProperties.getProviders().get(HXI_AUTH_PROVIDER);
         clearAuthHeaders(exchange);
-        exchange.getIn().setHeader(AUTHORIZATION, getBearerAuthHeader(HXI_AUTH_PROVIDER));
-        exchange.getIn().setHeader(ENVIRONMENT_KEY_HEADER, authProvider.getEnvironmentKey());
-        log.debug("Authorization :: {} authorization header added", HXI_AUTH_PROVIDER);
+        getHxpAuthHeaders().forEach((key, value) -> exchange.getIn().setHeader(key, value));
+        log.debug("Authorization :: {} authorization header added", HXP_AUTH_PROVIDER);
+    }
+
+    public Map<String, String> getHxpAuthHeaders()
+    {
+        return Map.ofEntries(getAuthHeader(HXP_AUTH_PROVIDER), getHxpEnvironmentHeader(), getHxpAppHeader());
+    }
+
+    public Map<String, String> getAlfrescoAuthHeaders()
+    {
+        return Map.ofEntries(getAuthHeader(ALFRESCO_AUTH_PROVIDER));
+    }
+
+    public Map.Entry<String, String> getAuthHeader(String providerId)
+    {
+        return Map.entry(AUTHORIZATION, getAuthHeaderValue(providerId));
+    }
+
+    protected Map.Entry<String, String> getHxpEnvironmentHeader()
+    {
+        return new AbstractMap.SimpleEntry<>(HXP_ENVIRONMENT_HEADER, ofNullable(authProperties.getProviders().get(HXP_AUTH_PROVIDER))
```

Comment:
```suggestion
        return new AbstractMap.SimpleEntry<>(
            HXP_ENVIRONMENT_HEADER,
            ofNullable(authProperties.getProviders().get(HXP_AUTH_PROVIDER))
                .map(AuthProperties.AuthProvider::getEnvironmentKey)
                .orElse(null)
        );
```

I'd suggest formatting this differently to make it easier to read.

Reply:
Great suggestion, thanks Domenico!